### PR TITLE
completes task 4

### PR DESCRIPTION
**Pull request by JonathanCrewe**
I wasn't sure if I should test for the exact article values or just toMatchObject(). 

Wanted to format the date value but ended up hard-coding it as I wasn't sure how to do it.

### CODE REVIEW

**Comment by JonathanCrewe** [comment]
Forgot to update endpoints.json. I'll include it with task 5.

File: __tests__/articles.test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+})
+
+
+describe("GET/api/articles/:article_id", () => {
+    test("400 - responds with a 400 Bad Request if id is an invalid type", async () => {
+        const {body}  = await request(app).get("/api/articles/BadID").expect(400)
+        expect(body.msg).toBe("Bad Request")
+    })
+
+    test("404 - responds with a 400 Bad Request if id is not in DB", async () => {
```
Comment (nc-aimee-muter) [inline __tests__/articles.test.js:24]: Test description needs updating.

File: __tests__/articles.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            article_img_url:
+              "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+          }
+
+
+        const {body}  = await request(app).get("/api/articles/1").expect(200)
+        const returnedArticle = body.article
+
+        expect(returnedArticle).toEqual(expectedArticle)
+    })
```
Comment (nc-aimee-muter) [inline __tests__/articles.test.js:47]: Testing looks good.

File: app.js [hunk 3]
```diff
@@ -1,16 +1,19 @@
 const express = require('express')
 const {getTopics} = require('./controllers/topics.controller.js')
 const {getEndpoints} = require('./controllers/endpoints.controller.js')
+const {getArticleById} = require('./controllers/articles.controller.js')
 
 
 // Set up. 
 const app = express()
 app.use(express.json())
```
Comment (nc-aimee-muter) [inline app.js:9]: You don't need this yet (app.use(express.json())) so please remove it until you do.

File: controllers/articles.controller.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        const articleId = parseInt(req.params.article_id)
+        const returnedArticle = await selectArticleById(articleId)
+        res.status(200).send({article: returnedArticle})
+    }
+    catch(err) {
+        next(err)
+    }
+}
+
+
```
Comment (nc-aimee-muter) [inline controllers/articles.controller.js:14]: Looks good!

File: models/articles.model.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    const articleResult = await db.query(queryStr, [id])
+
+    // If none returned - 404. 
+    if (articleResult.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Not Found" })
+    } 
+    
+    return articleResult.rows[0]    
+}
+
```
Comment (nc-aimee-muter) [inline models/articles.model.js:24]: Nice model function!

**Review by nc-aimee-muter — COMMENTED** [review]
Really nice code here!

**Pull request merged**